service/discovery/handlers.go

```diff
@@ -0,0 +1,229 @@
+package discovery
+
+import (
+	"net/http"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/iryonetwork/wwm/gen/discovery/models"
+	"github.com/iryonetwork/wwm/gen/discovery/restapi/operations"
+	"github.com/iryonetwork/wwm/storage/discovery"
+	"github.com/rs/zerolog"
+)
+
+// Handlers describes the actions supported by the discovery handlers
+type Handlers interface {
+	Query() operations.QueryHandler
+	Create() operations.CreateHandler
+	Update() operations.UpdateHandler
+	Delete() operations.DeleteHandler
+	Fetch() operations.FetchHandler
+	Link() operations.LinkHandler
+	ProxyLink() operations.LinkHandler
+	Unlink() operations.UnlinkHandler
+	ProxyUnlink() operations.UnlinkHandler
+	Authorizer() runtime.Authorizer
+	GetUserIDFromToken(token string) (*string, error)
+}
+
+type handlers struct {
+	service Service
+	logger  zerolog.Logger
+}
+
+func (h *handlers) Query() operations.QueryHandler {
+	return operations.QueryHandlerFunc(func(params operations.QueryParams, principal *string) middleware.Responder {
+		q := ""
```

Comment:
You could use `swag.StringValue` for this: [https://godoc.org/github.com/go-openapi/swag#StringValue](https://godoc.org/github.com/go-openapi/swag#StringValue)

Reply:
That makes a lot more readable

Reply:
Fixed